internal/stolonctl/pkg/crd/client.go

```diff
@@ -125,3 +124,104 @@ func (c *Client) List() (*StolonUpgradeList, error) {
 	}
 	return &result, nil
 }
+
+func (c *Client) get(name string) (*StolonUpgradeResource, error) {
+	var raw runtime.Unknown
+	err := c.Client.Get().
+		Namespace(c.namespace).
+		Resource(c.plural).
+		Name(name).
+		Do().
+		Into(&raw)
+	if err != nil {
+		return nil, rigging.ConvertError(err)
+	}
+
+	var result StolonUpgradeResource
+	if err := json.Unmarshal(raw.Raw, &result); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &result, nil
+}
+
+func (c *Client) update(obj *StolonUpgradeResource) (*StolonUpgradeResource, error) {
+	data, err := json.Marshal(obj)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var raw runtime.Unknown
+	err = c.Client.Put().
+		Namespace(c.namespace).
+		Resource(c.plural).
+		Name(obj.Name).
+		Body(data).
+		Do().
+		Into(&raw)
+	if err != nil {
+		return nil, rigging.ConvertError(err)
+	}
+
+	var result StolonUpgradeResource
+	if err := json.Unmarshal(raw.Raw, &result); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &result, nil
+}
+
+// MarkStepCompleted marks phase as completed
+func (c *Client) MarkStepCompleted(obj *StolonUpgradeResource, phaseName string) (*StolonUpgradeResource, error) {
+	var phases []StolonUpgradePhase
+	for _, phase := range obj.Spec.Phases {
+		if phaseName == phase.Name {
+			phase.Status = StolonUpgradeStatusCompleted
+			phase.FinishTimestamp = time.Now().UTC()
```

Comment:
This will always update the finish timestamp whenever `MarkStepCompleted` is called on the same phase - i.e. when the next step fails the process needs to be restarted - is this desired?
The same for all `MarkXXX` methods.